scipy/optimize/linprog_ip.py

```diff
@@ -0,0 +1,1994 @@
+# -*- coding: utf-8 -*-
+"""
+Created on Thu Feb 02 17:09:17 2017
+
+@author: Matt
+
+To do:
+multiple presolve
+questions for scipy-dev
+* This is written in Python 2. How should I check compatibility with Python 3?
+* Are the warnings OK? Do I need an option to disable them?
+* Is 100% pep8 compliance necessary?
+* How should unit tests be performed on both simplex and ip?
+* Can I leave multi-pass presolve to an update?
+* Can I leave sparse presolve to an update?
+* Do I need to implement a callback interface?
+* There are some changes to be made to simplex; it should use the ip presolve
+ routine. Can that be left to an update?
+* How do I document options not common with simplex?
+* How do I use _check_unknown_options?
+* Should I expose the choice of sparse matrix format to the user?
+* Should I expose the choice of sparse matrix solve pivot rule to the user?
+* there are many different tolerances. For well-scaled problems, they will
+    work just fine and I don't personally see a need to expose them all.
+    Should I anyway?
+* What should the file structure be? Can I keep things in separate files?
+* Should interior-point be the default?
```

Comment:
It is faster. But yes, it could break some codes due to the nature of the solution returned (potentially not a vertex of the polytope defined by the constraints) so initially I agree that it should not be the default.

However, in addition to the suggested modifications to `_linprog.py`, I temporarily made 'interior-point' the default for ease of testing. I'll switch back when I understand your suggestion about how to run common tests on both the simplex and interior point methods.